src/server/pps/server/master.go

```diff
@@ -106,29 +113,55 @@ func (a *apiServer) master() {
 	})
 }
 
+func (a *apiServer) setPipelineFailure(ctx context.Context, pipelineInfo *pps.PipelineInfo) error {
+	// Set pipeline state to running
+	_, err := col.NewSTM(ctx, a.etcdClient, func(stm col.STM) error {
+		pipelineName := pipelineInfo.Pipeline.Name
+		pipelines := a.pipelines.ReadWrite(stm)
+		pipelineInfo := new(pps.PipelineInfo)
+		if err := pipelines.Get(pipelineName, pipelineInfo); err != nil {
+			return err
+		}
+		pipelineInfo.State = pps.PipelineState_PIPELINE_FAILURE
+		pipelines.Put(pipelineName, pipelineInfo)
+		return nil
+	})
+	return err
+}
+
```

Comment:
We don't automatically recover (since it's unclear how to recover if k8s won't let you create workers), and there's no meaningful action that the user can take other than maybe rebooting the cluster and hope that something happens.  This PR mostly tries to solve the problem that currently if we fail to create workers for a pipeline, all subsequent pipeline creations are ignored.